Dockerfile

```diff
@@ -93,19 +93,63 @@ RUN pip install -e . --no-deps
 ###################
 ## Frontend Base ##
 ###################
-FROM node:16-slim as frontend
+FROM node:16-alpine as frontend
+
+RUN apk add --no-cache libc6-compat
+# Build the frontend clients
```

Comment:
I'm not convinced that we want to be building PC in this dockerfile since pc is a separate app and already has its own dockerfile. to me this Dockerfile is just for `fides` and should be pretty similar to what the python package can serve up, but with a few helpful stages. maybe I'm missing something and can still be convinced though :)

Reply:
So PC is still deployed as a separate app, which is what we do in our `prod_pc` stage of this Dockerfile. 

With my changes, both PC and admin-ui require the same `frontend` base target now, which includes everything in `clients`. This is because using Turborepo, we need to install and build everything from the base `clients` folder. The benefit of using the same Dockerfile for both PC and admin-ui is that we get to re-use this cached `frontend` image whenever we run or deploy both PC and admin-ui, saving build time.

Could you maybe explain more about the downfalls of having this Dockerfile build PC too?

Reply:
While generally I do like separation of concerns, I think there is enough overlap here to justify combining the Dockerfiles if it means faster build/dev times and is well documented!

Reply:
ah yeah, I see your point. I keep forgetting that they will all share `node_modules` now. 

although I imagine you could do probably install only the admin ui dependencies, then run just the admin ui here? admin-ui and privacy-center share some dependencies but admin-ui has a lot more than privacy-center does, so if you're just trying to run the privacy center via nox, you'll end up needing to install a bunch of packages you don't need. so it's kind of like yes, they can share the same `frontend` stage, but only because the `frontend` stage is installing more things

that said, I rarely run the frontends through nox/docker since the hot reloading is slower. so we likely won't notice this kind of thing, but just want to call it out so we are aware!

all that to say, let's use what you have here and we can keep iterating if needed :)

Reply:
Ah I see what you're saying @allisonking , and that would be ideal if we could just install the dependencies for 1 workspace e.g. `privacy-center`. I'm digging in docs a bit, and everything seems to recommend installing everything at root https://turbo.build/repo/docs/handbook/workspaces#managing-workspaces, and I'm unsure if there's a way to filter out specific workspaces here. This is because install itself uses `npm`, not turborepo. Turbo itself is capable of filtering workspaces.

Reply:
There's some really cool & interesting tools for optimizing the Docker builds with Turborepo. The docs for this are well written too: https://turbo.build/repo/docs/handbook/deploying-with-docker

The TL;DR is that it's entirely possible to use turborepo to:
* "prune" the node_modules / dependencies / etc for a single app in the monorepo (using `turbo prune`)
* copy over the build dependencies to a Docker layer (optimally) using `turbo prune --docker`
* trace the exact built files needed for the production server
* etc.

I made a "note to self" about these in a comment here: https://github.com/ethyca/fides/blob/80ec6695f1f636a7e8a32935a411b8f9b69769b8/Dockerfile#L134

My plan is to just farm this out into a separate follow-up ticket though. Let's get the monorepo working 👍